core/src/test/java/org/elasticsearch/index/IndexServiceTests.java

```diff
@@ -187,11 +189,18 @@ protected String getThreadPool() {
                 return ThreadPool.Names.GENERIC;
             }
         };
+
+        BiFunction<AtomicReference<CountDownLatch>, CountDownLatch, CountDownLatch> swapAndReturn = (ref, newLatch) -> {
+            CountDownLatch downLatch = ref.get();
+            ref.set(newLatch);
+            return downLatch;
+        };
         latch.get().await();
-        latch.set(new CountDownLatch(1));
+        swapAndReturn.apply(latch, new CountDownLatch(1));
         assertEquals(1, count.get());
-        latch2.get().countDown();
-        latch2.set(new CountDownLatch(1));
+        // here we need to swap first before we let it go otherwise threads might be very fast and run that task twice due to
+        // random exception and the schedule interval is 1ms
+        swapAndReturn.apply(latch2, new CountDownLatch(1)).countDown();
```

Comment:
If my previous comment was right :)  And this is now the only use of the new lambda, maybe don't use a lambda and inline the code here?